iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift

```diff
@@ -292,117 +292,115 @@ class SwipeableInputAccessoryView: View, AttachmentViewControllerDelegate, UIGes
+        guard self.shouldHandlePan() else { return }
+
+        // How far the preview view can be dragged left or right.
+        let maxXOffset: CGFloat = 100
+        // How far the preview view can be dragged vertically
+        let maxYOffset: CGFloat = SwipeableInputAccessoryView.maxHeight.half
+        let panOffset = pan.translation(in: nil)
+        // The percentage of the max y offset that the preview view has been dragged.
+        let progress = clamp(-panOffset.y/maxYOffset, 0, 1)
+
+        switch pan.state {
+        case .possible:
+            break
+        case .began:
+            self.handlePanBegan()
+        case .changed:
+            self.handlePanChanged(withOffset: panOffset, maxXOffset: maxXOffset, maxYOffset: maxYOffset)
+        case .ended:
+            self.handlePanEnded(withProgress: progress)
+        case .cancelled, .failed:
+            self.handlePanFailed()
+        @unknown default:
+            break
+        }
+    }
+
     func shouldHandlePan() -> Bool {
-        let object = SendableObject(kind: self.currentMessageKind, context: self.currentContext, previousMessage: self.editableMessage)
-        self.sendableObject = object
+        let object = SendableObject(kind: self.currentMessageKind,
+                                    context: self.currentContext,
+                                    previousMessage: self.editableMessage)
+
         return object.isSendable
     }
 
-    func panDidBegin() {
+    private func handlePanBegan() {
+        let object = SendableObject(kind: self.currentMessageKind,
+                                    context: self.currentContext,
+                                    previousMessage: self.editableMessage)
+        self.sendableObject = object
+
+        self.attachmentView.alpha = 0
+        self.textView.alpha = 0
+
+        // Initialize the preview view for the user to drag up the screen.
+        self.previewView = PreviewMessageView()
+        self.previewView?.frame = self.inputContainerView.frame
         self.previewView?.set(backgroundColor: self.currentContext.color)
         self.previewView?.messageKind = self.currentMessageKind
-    }
+        self.addSubview(self.previewView!)
 
-    func previewAnimatorDidEnd() {
-        self.reset()
-        self.previewView?.removeFromSuperview()
+        self.initialPreviewOrigin = self.previewView?.origin
+    }
 
-        if let object = self.sendableObject {
-            self.delegate.swipeableInputAccessory(self, didConfirm: object)
-        }
+    private func handlePanChanged(withOffset offset: CGPoint, maxXOffset: CGFloat, maxYOffset: CGFloat) {
+        guard let initialPosition = self.initialPreviewOrigin else { return }
 
-        self.sendableObject = nil
+        let offsetX = clamp(offset.x, -maxXOffset, maxXOffset)
+        let offsetY = clamp(offset.y, -maxYOffset, 0)
+        self.previewView?.origin = initialPosition + CGPoint(x: offsetX, y: offsetY)
```

Comment:
Move the view around based on the pan gesture, but limit how far you can drag in any direction.